src/Orleans/Runtime/GrainMethodInvocationExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.CodeGeneration;
+using Orleans.Concurrency;
+
+namespace Orleans.Runtime.GrainMethodInvocationExtensions
+{
+    public static class GrainMethodInvocationExtensions
+    {
+        /// <summary>
+        /// Invokes a method of a grain interface is one-way fashion so that no response message will be sent to the caller.
+        /// </summary>
+        /// <typeparam name="T">Grain interface</typeparam>
+        /// <param name="grainReference">Grain reference which will be copied and then a call executed on it</param>
+        /// <param name="grainMethodInvocation">Function that should invoke grain method and return resulting task</param>
+        public static void InvokeOneWay<T>(this T grainReference, Func<T, Task> grainMethodInvocation) where T : class, IGrain
+        {
+            // double cast is safe, as it can be invoked by user only on grain references
+            var oneWayGrainReferenceCopy = new GrainReference((GrainReference) (object)grainReference, InvokeMethodOptions.OneWay).Cast<T>();
+
```

Comment:
We could also loosen the generic constraint to be `IAddressable` instead of grain - no big deal either way, but it might be useful for implementing some lower-level protocols using `GrainService`/`SystemTarget` at some point.